clusterloader2/pkg/measurement/common/network/performance-measurement-helper.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package network
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+func (npm *networkPerformanceMeasurement) validate(config *measurement.Config) error {
+	var err error
+	if npm.testDuration, err = util.GetInt(config.Params, "duration"); err != nil {
+		return err
+	}
+	if npm.protocol, err = util.GetString(config.Params, "protocol"); err != nil {
+		return err
+	}
+	if npm.protocol != protocolTCP && npm.protocol != protocolUDP && npm.protocol != protocolHTTP {
+		return fmt.Errorf("invalid protocol , supported ones are TCP,UDP,HTTP")
+	}
+	if err = npm.validatePodConfiguration(config); err != nil {
+		return err
+	}
+	npm.podRatioType = npm.getPodRatioType()
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) validatePodConfiguration(config *measurement.Config) error {
+	var err error
+	if npm.numberOfClients, err = util.GetInt(config.Params, "numberOfClients"); err != nil {
+		return err
+	}
+	if npm.numberOfServers, err = util.GetInt(config.Params, "numberOfServers"); err != nil {
+		return err
+	}
+	if npm.numberOfClients <= 0 || npm.numberOfServers <= 0 {
+		return fmt.Errorf("invalid number of client or server pods")
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) getPodRatioType() string {
+	if npm.numberOfClients == npm.numberOfServers && npm.numberOfClients == 1 {
+		return oneToOne
+	}
+	if (npm.numberOfClients > npm.numberOfServers) && npm.numberOfServers == 1 {
+		return manyToOne
```

Comment:
This case doesn't seem to be handleded anywhere....

Reply:
Yes, we do have future testcases where we want to code N:1 ratio. So we have included this condition. If required , will remove it for time being.